From 9dcff22fd71f34809b3c054a36677d1371cf799d Mon Sep 17 00:00:00 2001
From: Jerome Brunet <jbrunet@baylibre.com>
Date: Tue, 6 Sep 2022 16:32:54 +0200
Subject: [PATCH] arm64: dts: amlogic: add AO rti sticky register sram

Add the general purpose scratch registers AO_RTI_STICKY as
an sram device.

Signed-off-by: Jerome Brunet <jbrunet@baylibre.com>
Upstream-Status: Pending
---
 arch/arm64/boot/dts/amlogic/meson-axg.dtsi        | 8 ++++++++
 arch/arm64/boot/dts/amlogic/meson-g12-common.dtsi | 8 ++++++++
 arch/arm64/boot/dts/amlogic/meson-gx.dtsi         | 8 ++++++++
 3 files changed, 24 insertions(+)

diff --git a/arch/arm64/boot/dts/amlogic/meson-axg.dtsi b/arch/arm64/boot/dts/amlogic/meson-axg.dtsi
index 768d0ed78dbe..c384591cee7f 100644
--- a/arch/arm64/boot/dts/amlogic/meson-axg.dtsi
+++ b/arch/arm64/boot/dts/amlogic/meson-axg.dtsi
@@ -1657,6 +1657,14 @@ mux {
 				};
 			};
 
+			rti_sticky_AO: sram@130 {
+				compatible = "mmio-sram";
+				reg = <0x0 0x130 0x0 0x10>;
+				ranges = <0 0x0 0x130 0x10>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+			};
+
 			sec_AO: ao-secure@140 {
 				compatible = "amlogic,meson-gx-ao-secure", "syscon";
 				reg = <0x0 0x140 0x0 0x140>;
diff --git a/arch/arm64/boot/dts/amlogic/meson-g12-common.dtsi b/arch/arm64/boot/dts/amlogic/meson-g12-common.dtsi
index ff68b911b729..9b6dd7cd901a 100644
--- a/arch/arm64/boot/dts/amlogic/meson-g12-common.dtsi
+++ b/arch/arm64/boot/dts/amlogic/meson-g12-common.dtsi
@@ -2023,6 +2023,14 @@ cec_AO: cec@100 {
 				status = "disabled";
 			};
 
+			rti_sticky_AO: sram@130 {
+				compatible = "mmio-sram";
+				reg = <0x0 0x130 0x0 0x10>;
+				ranges = <0 0x0 0x130 0x10>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+			};
+
 			sec_AO: ao-secure@140 {
 				compatible = "amlogic,meson-gx-ao-secure", "syscon";
 				reg = <0x0 0x140 0x0 0x140>;
diff --git a/arch/arm64/boot/dts/amlogic/meson-gx.dtsi b/arch/arm64/boot/dts/amlogic/meson-gx.dtsi
index 2673f0dbafe7..f3921711ae9a 100644
--- a/arch/arm64/boot/dts/amlogic/meson-gx.dtsi
+++ b/arch/arm64/boot/dts/amlogic/meson-gx.dtsi
@@ -468,6 +468,14 @@ cec_AO: cec@100 {
 				status = "disabled";
 			};
 
+			rti_sticky_AO: sram@130 {
+				compatible = "mmio-sram";
+				reg = <0x0 0x00130 0x0 0x10>;
+				ranges = <0 0x0 0x130 0x10>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+			};
+
 			sec_AO: ao-secure@140 {
 				compatible = "amlogic,meson-gx-ao-secure", "syscon";
 				reg = <0x0 0x140 0x0 0x140>;
-- 
2.43.0

